### ISSUE TITLE
A non-numeric value encountered in SUITECRM/include/ListView/ListViewSubPanel.php on line [264|546]

### ISSUE THREAD

**Issue by jobst** [salesagility/SuiteCRM#3576]
A non-numeric value encountered in SUITECRM/include/ListView/ListViewSubPanel.php on line 546.

#### Issue

When checking the value of "$current_offset" when the error (subject line) appears its empty/null.
You cannot add 1 to a null/empty value.

#### Expected Behavior

Its probably an oversight of the person writing that code not expecting the person using the function NOT checking whether the passed value is actually an integer value.

#### Actual Behavior

error and stack trace shows up in the log.

#### Possible Fix

Replace this line (264)

           $offset = $this->getOffset($html_varName);

with this line

           $offset = empty($this->getOffset($html_varName)) ? 0:$this->getOffset($html_varName);

Replace this line (546)

            $start_record = $current_offset + 1;

with these three lines

            if( !isset($current_offset) || empty($current_offset) )                                                                    
              $current_offset=0;
            $start_record = $current_offset + 1;

#### Steps to Reproduce

1. open a ANY list of any subpanel
1a. in my case a user that had no role and security management defined
2. if the list that was opened has no member the error arises

#### Context

Its ugly code and throws exceptions everywhere a member list of any subpanel is zero.
It will also slow down the processing as the exceptions needs to be dealt with.

#### Your Environment

* SuiteCRM Version used: 7.8.2
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): N/A
* Environment name and version (e.g. MySQL, PHP 7): N/A
* Operating System and version (e.g Ubuntu 16.04): N/A

**Comment by chris001** [comment]
Good find!  Make a PR.

**Comment by daniel-samson** [comment]
See https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM for more information about contributing to SuiteCRM.

**Comment by jobst** [comment]
@chris001 I have a problem submitting this fix - when creating a PR for this it seems wanting to submit my previous PR's as well. 
How can I stop this from happening?

**Comment by chris001** [comment]
@jobst It's wanting to submit your previous PR's as well, because your fork's hotfix branch is a copy of each file in this repo, at the time you forked this repo.  You edited your copies of the files, fixed bugs, made PRs of them.  However since those PRs weren't merged yet, a PR made today from your hotfix branch of this repo is reflecting all your unmerged changed files and bugs you have fixed since you forked this repo.  The answer is to make a new branch in your fork, name it after this issue, have it be a copy of today's version of the hotfix branch, and fix the bug in that new branch in your fork.
https://git-scm.com/book/en/v2/Git-Branching-Rebasing

**Comment by jobst** [comment]
I have them all in different branches named after what they do, e.g.
 - jobst-ListViewSubPanel_current_offset-1 
 - jobst-ListViewSmarty-1 
 - jobst-rest-patch-1-2
and so on ... I currently stopped adding other as I have to sort out my problems.

I am thinking of wiping it all and starting fresh but then I do not know what happens to the changes I have already PR'ed (for example the rest interface).

I can write bash scripts most people wouldn't even understand, I can control some hardware I created from this workstation via a web-interface I wrote, I am looking after a full blown online training site I wrote with the help from some other developers, I look after a LAN/WAN and some servers I own in a data center, I run named, apache, sendmail and many other services plus everything is supervised by nagios ....

I understand what GIT does and how complicated it is (due to many contributors working in parallel) - but the GIT interface has me stumped - even Git's web-interface - it is not intuitive (at least for me).  It requires too much detailed knowledge for what I want to do: quickly suggest some bloody code/bug/enhancement fixes (and the 20+ other I still have here) ....
